app/controllers/api/v1/audios_controller.rb

```diff
@@ -46,7 +46,7 @@ def create
 
         # Note(lyang): When replacing an existing file with a differently named file,
         # we need to update the :filename primary key.
-        if params[:originalFilename]
+        if params[:originalFilename] && (params[:originalFilename] != params[:flowFilename])
```

Comment:
So this means:

If the original file exists and the new filename is different from the old filename
--> change the filename of this record to the new filename?

Reply:
Yep, that's right. The old (buggy) behavior was that if you uploaded to an existing audio but with a different filename, instead of overwriting the original filename, it would just create essentially a duplicate record, but with a different filename.

Reply:
OHHHH! That makes sense! Okay, looks good! :shipit:  when ready